File: 1-twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const obj = {}
    for (let i = 0; i < nums.length; i++) {
      const remainder = target - nums[i];
      if (obj[remainder] !== undefined) {
        return [i, obj[remainder]]
      }

      obj[nums[i]] = i;
    }
};

File: two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        d = {}
        for i, n in enumerate(nums):
            if n in d:
                return([d[n], i])
            else:
                d[target - n] = i


if __name__ in "__main__":
    s = Solution()
    assert s.twoSum([2, 7, 11, 15], 9) == [0, 1]
    assert s.twoSum([3, 2, 4], 6) == [1, 2]
    assert s.twoSum([3, 3], 6) == [0, 1]